src/oracc_cmd.ts

```diff
@@ -0,0 +1,29 @@
+import * as child_process from 'child_process';
+import { homedir } from 'os';
+import * as process from 'process';
+import { promisify } from 'util';
+import * as vscode from 'vscode';
+
+const execFile = promisify(child_process.execFile);
+
+function run_cmd(file: string,
+                 args: string[] = [],
+                 options: Object = {
+                     cwd: homedir(),
+                     env: {
+                         ...process.env,
+                         ...{PATH: '/usr/local/oracc/bin:' + process.env.PATH},
+                     },
+                 }) {
+    return execFile(file, args, options);
+}
+
+export async function run_oracc(action: string) {
+    try {
+        const { stdout } = await run_cmd('oracc', [action]);
+        vscode.window.showWarningMessage(`${stdout}`);
+    } catch (err) {
+        vscode.window.showErrorMessage(`Could not run ${action}: ${err}`);
```

Comment:
The error message can be improved.  At the moment this `catch` is catching both command failing because the `oracc` command couldn't be found (I think we should deal with this case specially, for example suggesting to make sure to run the command on the server) and command failing because returning any other non-zero code.